apps/ws/src/socket/ws.gateway.ts

```diff
@@ -129,9 +129,28 @@ export class WSGateway implements OnGatewayConnection, OnGatewayDisconnect {
     socket.disconnect();
   }
 
-  private async getActiveConnections(socket: Socket, subscriberId: string) {
-    const activeSockets = await socket.in(subscriberId).fetchSockets();
+  async gracefulShutdown(): Promise<void> {
+    try {
+      if (!this.server) {
+        Logger.error('WS server was not initialized while executing shutdown', LOG_CONTEXT);
+
+        return;
+      }
+
+      Logger.log('Closing WS server for incoming new connections', LOG_CONTEXT);
+      this.server.close();
+
+      Logger.log('Disconnecting active sockets connections', LOG_CONTEXT);
+      this.server.sockets.disconnectSockets();
```

Comment:
At the moment I have not implemented disconnection with jitter due to a lack of data, i think we should make a small discovery on how many connections we have and how we want to split them unless someone is familiar with general best practices that will fit our needs that we could implement here.